packages/application-shell/src/components/application-page-title/application-page-title.tsx

```diff
@@ -0,0 +1,76 @@
+import { useLayoutEffect } from 'react';
+import { useLocation } from 'react-router-dom';
+import upperFirst from 'lodash/upperFirst';
+
+type TApplicationPageTitleProps = {
+  /**
+   * Prepend additional title parts to the default page title.
+   * The parts are concatenated with a `-` separator and might be truncated
+   * in the middle if each one exceeds 24 characters length.
+   *
+   * The default page title has always the format:
+   *   <application_identifier> - <project_key> - Merchant Center
+   *
+   * Overwriting the default page title is recommended in detail pages
+   * where there is a human-readable resource identifier, for example a product name.
+   *
+   * @example
+   *
+   * <ApplicationPageTitle additionalParts={['T-Shirt red']} />
+   * // T-Shirt red - Products - my-shop - Merchant Center
+   */
+  additionalParts: string[];
+};
+
+const maxTitleCharLength = 24;
+const staticPaths = ['account', 'login'];
+
+const defaultProps: Pick<TApplicationPageTitleProps, 'additionalParts'> = {
```

Comment:
nit: perhaps we don't need to use `Pick` helper anymore since the type only has one prop.